_includes/list-partners.html

```diff
@@ -0,0 +1,17 @@
+{% for agency in project.partner %}
+  {% capture agencyname %}{{ agency | slugify }}{% endcapture %}
+  {% capture partnerurl %}{% for partner in site.data.partners %}{% if partner[0] == agencyname %}{{ partner[1].url }}{% else %} {% endif %}{% endfor %}{% endcapture %}
+
+  {% if include.style == "list" %}
+    {% unless partnerurl == " " %}
+      <li><a href="{{ partnerurl }}">{{ agency }}</a></li>
+    {% else %}
+      <li>{{agency}}</li>
+    {% endunless %}
+  {% elsif include.style == "strong" %}
+    {% unless partnerurl == " " %}
+      <strong><a href="{{partnerur l}}">{{agency}}{% unless forloop.rindex == 1 %}, {% endunless%}</a></strong>
+    {% else %}
+      <strong>{{ agency }}</strong>
+  {% endif %}
+{% endfor %}
```

Comment:
As I'm sure, you know, this'll inject a bunch of newlines and spaces into the output, as Liquid preserves them _all_. Up to you if you want to hoist some of these expressions onto a single line to avoid such bloat. Ugly to read, but some of the PIF website pages @stroupaloop worked on were O(400K) until he did this.